src/main/scala/cromwell/engine/backend/io/package.scala

```diff
@@ -0,0 +1,58 @@
+package cromwell.engine.backend
+
+import java.nio.file._
+
+import cromwell.engine.backend.io.filesystem.gcs.{ContentTypeOption, GcsFileSystem, GcsFileSystemProvider, NioGcsPath}
+
+import scala.util.{Success, Try}
+
+package object io {
+  object PathString {
+    val defaultGCSFileSystem = GcsFileSystem.defaultGcsFileSystem
+    val defaultFileSystem = FileSystems.getDefault
+    private val defaultFileSystemsList = List(defaultGCSFileSystem, defaultFileSystem)
+
+    implicit class UriString(val str: String) extends AnyVal {
+
+      def isGcsUrl: Boolean = str.startsWith("gs://")
+      def isUriWithProtocol: Boolean = "^[a-z]+://".r.findFirstIn(str).nonEmpty
+
+      def toPath(implicit fss: List[FileSystem] = defaultFileSystemsList): Path = {
```

Comment:
I'm not fond of the name `toPath` but I also can't think of a better name...

I'm also curious why everything is done with implicits... I know I tend to be more against implicits than other people, so maybe this isn't a cause for concern... I can sometimes get lost figuring out where things are actually being created (most specifically, this `List[FileSystem]` parameter, the `UriString` implicit class is less of an issue for me)